test/sim/abilities/wonderguard.js

```diff
@@ -43,4 +43,12 @@ describe('Wonder Guard', function () {
 		battle.setPlayer('p2', {team: [{species: "Reshiram", ability: 'turboblaze', moves: ['fusionflare']}]});
 		assert.hurts(battle.p1.active[0], () => battle.makeChoices('move sleeptalk', 'move fusionflare'));
 	});
+
+	it('should allow first turn of Sky Drop but not the second', function () {
+		battle = common.createBattle();
+		battle.setPlayer('p1', {team: [{species: "Smeargle", ability: 'wonderguard', moves: ['sleeptalk']}]});
+		battle.setPlayer('p2', {team: [{species: "Aerodactyl", ability: 'owntempo', moves: ['skydrop']}]});
+		battle.makeChoices('move sleeptalk', 'move skydrop');
```

Comment:
You should probably have Smeargle use a move with a side effect (like Howl), so you can make sure it didn't take effect.

Reply:
This is, after all, the most important part of this test: You need to write a test that fails before your test, and succeeds now.

This is one of the most important parts of writing tests: Fixing confirmation bias. It's not enough for a test to pass for things you want, it also needs to fail for things you don't want.

Reply:
Okay, that's good to know for the future